app/src/main/java/com/wut/indoornavigation/logic/graph/impl/GraphImpl.java

```diff
@@ -0,0 +1,241 @@
+package com.wut.indoornavigation.logic.graph.impl;
+
+
+import android.os.Build;
+import android.support.annotation.NonNull;
+import android.support.annotation.RequiresApi;
+
+import com.wut.indoornavigation.logic.graph.Graph;
+import com.wut.indoornavigation.logic.graph.HeuristicFunction;
+import com.wut.indoornavigation.logic.graph.UnionFind;
+import com.wut.indoornavigation.logic.graph.models.Edge;
+import com.wut.indoornavigation.logic.graph.models.Vertex;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.PriorityQueue;
+
+
+public class GraphImpl implements Graph {
+    private List<Vertex> vertices;
+    private Map<Vertex, List<Edge>> edges;
+
+
+    public GraphImpl(@NonNull List<Vertex> vertices) {
+        this.vertices = new ArrayList<>();
+        this.edges = new HashMap<>();
+
+        int verticesCount = vertices.size();
+        for(int i=0; i < verticesCount; i++){
+            this.vertices.add(vertices.get(i));
```

Comment:
wystarczy ze zamiast this.vertices  = new ArrayList().... i tego zrobisz this.vertices.addAll(vertices);